pod/pod.go

```diff
@@ -6,10 +6,50 @@ import (
 	"syscall"
 )
 
+func reapChildProcesses() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGCHLD)
+
+	for {
+		// Block until SIGCHLD signal is received.
+		<-c
+		reapChildProcessesHelper()
+	}
+}
+
+func reapChildProcessesHelper() {
+	for {
+		// Pid -1 means to wait for any child process.
+		// With syscall.WNOHANG option set, function will
+		// not block and will exit immediatelly if no child
+		// process has exited.
+		_, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+
+		switch err {
+		case nil:
+			// One or more child process still exist,
+			// but have not yet changed state. Try again.
+		case syscall.EINTR:
+			// Got interrupted. Shouldn't happen with WNOHANG option,
+			// but it is better to handle it anyway and try again.
+		case syscall.ECHILD:
+			// No more child processes to reap. We can return and wait
+			// for another SIGCHLD signal.
+			return
+		default:
+			// Some other unexpected error. Return and wait for
```

Comment:
log the error here

Reply:
Done. Please let me know if log.Println(err) is ok, or there is some other standard for logging.

Reply:
Yeah, `log` package is fine, but we should log to stderr with a prefix. So, something like:
```
l := log.New(os.Stderr, "reaper: ", log.LstdFlags|log.Lmicroseconds)
l.Printf("Unexpected error waiting for child processes: %v", err)
```
You could create the logger outside and pass it down. 
```

Reply:
Done